File: backend/Flowline.Domain/Entities/Project.cs
namespace Flowline.Domain.Entities;

public sealed record Project
{
    public required Guid Id { get; init; }
    public required Guid UserId { get; init; }
    public required string Name { get; init; }
    public required string Color { get; init; }
    public required bool IsArchived { get; init; }
    public required DateTime CreatedAt { get; init; }
}

File: frontend/src/types/project.ts
export interface Project {
  id: string;
  userId: string;
  name: string;
  color: string;
  isArchived: boolean;
  createdAt: string;
}

export interface CreateProjectRequest {
  userId: string;
  name: string;
  color: string;
}

export interface UpdateProjectRequest {
  userId: string;
  name?: string;
  color?: string;
  isArchived?: boolean;
}
